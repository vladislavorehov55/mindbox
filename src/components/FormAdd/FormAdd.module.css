
.input {
    width: 100%;
    height: 55px;
    padding: 5px;
    outline: none;

}