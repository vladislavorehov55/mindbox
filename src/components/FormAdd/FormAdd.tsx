import React, {FC, FormEvent, useState} from 'react';
import {ITaskItem} from '../../intefaces';
import {TaskType} from '../../enums';
import styles from './FormAdd.module.css';

type FormAddProps = {
  addTask: (newTaskItem: ITaskItem) => void
}
const FormAdd: FC<FormAddProps> = ({addTask}) => {

  const [value, setValue] = useState<string>('')
  const inputChangeHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
    setValue(e.target.value)
  }

  const submitHandler = (e: FormEvent) => {
    e.preventDefault()
    if (value) {
      addTask({id: Date.now(), text: value, type: TaskType.active})
    }
    setValue('')
  }

  return (
    <form onSubmit={submitHandler} role='form-add'>
      <input type="text" placeholder='What needs to be done?'
             value={value}
             onChange={inputChangeHandler}
             className={styles.input}
      />
    </form>
  );
};

export default FormAdd;