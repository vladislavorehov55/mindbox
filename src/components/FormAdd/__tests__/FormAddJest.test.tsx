import React from 'react';
import {mount, ReactWrapper} from 'enzyme';
import FormAdd from "../FormAdd";

let mockValue = 'Buy a cake';
const mockSetValue = jest.fn();
jest.mock('react', () => ({
    ...jest.requireActual('react'),
    useState: () => ([mockValue, mockSetValue])
}))

describe('FormAdd component', () => {
    let wrapper: ReactWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;

    const addTask = jest.fn();

    const getWrapper = () => mount(<FormAdd addTask={addTask}/>);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = getWrapper();
    });

    it('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })

    it('should call setValue with correct todo', () => {
        const todo = 'Watch movie';
        const input = wrapper.find('input')
        input.simulate('change', { target: { value: todo}});
        expect(mockSetValue).toHaveBeenCalledWith(todo);
    })

    it('should call addTask and setValue', () => {
        wrapper.find('form').simulate('submit');
        expect(addTask).toHaveBeenCalled();
        expect(mockSetValue).toHaveBeenCalledWith('');
    })

    it('should not call addTask', () => {
        mockValue = '';
        wrapper = getWrapper();
        wrapper.find('form').simulate('submit');
        expect(addTask).not.toHaveBeenCalled();
    })
})
