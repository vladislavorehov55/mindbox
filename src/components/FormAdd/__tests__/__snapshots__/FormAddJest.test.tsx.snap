// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormAdd component should render correctly 1`] = `
<FormAdd
  addTask={[MockFunction]}
>
  <form
    onSubmit={[Function]}
    role="form-add"
  >
    <input
      onChange={[Function]}
      placeholder="What needs to be done?"
      type="text"
      value="Buy a cake"
    />
  </form>
</FormAdd>
`;
